pycloudlib/azure/instance.py

```diff
@@ -30,7 +31,16 @@ def wait_for_delete(self):
 
     def wait_for_stop(self):
         """Wait for instance stop."""
-        raise NotImplementedError
+        for _ in range(100):
+            power_state = self._client.virtual_machines.get(
+                resource_group_name=self._instance['rg_name'],
+                vm_name=self.name,
+                expand='instanceView'
+            ).instance_view.statuses[1].display_status
+            if power_state == 'VM stopped':
+                return
+            time.sleep(1)
+        raise TimeoutError
```

Comment:
Maybe we could alert the user the context of the TimeoutError, like indicating that it was during a `wait_for_stop`  operation